scipy/ndimage/_ni_support.py

```diff
@@ -95,3 +96,23 @@ def _get_output(output, input, shape=None, complex_output=False):
     elif complex_output and output.dtype.kind != 'c':
         raise RuntimeError("output must have complex dtype")
     return output
+
+
+def _check_axes(axes, ndim):
+    if axes is None:
+        return tuple(range(ndim))
+    elif numpy.isscalar(axes):
+        axes = (operator.index(axes),)
+    elif isinstance(axes, Iterable):
+        for ax in axes:
+            axes = tuple(operator.index(ax) for ax in axes)
+            if ax < -ndim or ax > ndim - 1:
+                raise ValueError(f"specified axis: {ax} is out of range")
+        axes = tuple(ax % ndim if ax < 0 else ax for ax in axes)
+    else:
+        raise ValueError(
+            "axes must be an integer, iterable of integers or None"
+        )
```

Comment:
Consider the Oxford comma to avoid confusion between "(iterable of integers) or (None)" and "iterable of (integers or None)". 

You're welcome to maintain the formatting if you prefer, but while I'm at it, I might as well suggest:
```suggestion
        message = "axes must be an integer, iterable of integers, or None"
        raise ValueError(message)
```